include/charmlite/core/collection.hpp

```diff
@@ -103,41 +104,55 @@ namespace cmk {
             auto* rec = msg->has_combiner() ? nullptr : record_for(ep.entry);
             // NOTE ( the lifetime of this variable is tied to the message! )
             auto& idx = ep.chare;
-            auto pe = this->locmgr_.pe_for(idx);
-            // TODO ( temporary constraint, elements only created on home pe )
-            if (rec && rec->is_constructor_ && (pe == CmiMyPe()))
+            auto home_pe = this->locmgr_.pe_for(idx);
```

Comment:
A few things:
1. This should be something more like "if (createhere || homePe(idx) == myPe)"
2. You can change the name of the existing `pe_for` method to `home_pe`, then add `lookup` which reports the last known location. Then it'll match Charm++ :)
3. When you do that, refactor the parts of the code that use `pe_for` to use the right method. Most should change to `lookup`.
4. For node/group locmgrs, `lookup` will be the same as `home_pe`.